core/src/main/java/org/apache/carbondata/core/util/TrashUtil.java

```diff
@@ -152,46 +153,77 @@ public static void copyFilesToTrash(List<String> filesToCopy,
    * The below method deletes timestamp subdirectories in the trash folder which have expired as
    * per the user defined retention time
    */
-  public static void deleteExpiredDataFromTrash(String tablePath) {
+  public static long[] deleteExpiredDataFromTrash(String tablePath, Boolean isDryRun) {
     CarbonFile trashFolder = FileFactory.getCarbonFile(CarbonTablePath
         .getTrashFolderPath(tablePath));
+    long sizeFreed = 0;
+    long trashFolderSize = 0;
     // Deleting the timestamp based subdirectories in the trashfolder by the given timestamp.
     try {
       if (trashFolder.isFileExist()) {
+        trashFolderSize = FileFactory.getDirectorySize(trashFolder.getAbsolutePath());
```

Comment:
i think you should not calculate size when the statistics is false, so calculate only when it is true else it will affect clean files time, please take care not to impact time of clean files

Reply:
done